controllers/vspherevm_controller.go

```diff
@@ -324,7 +325,19 @@ func (r vmReconciler) reconcileNormal(ctx *context.VMContext) (reconcile.Result,
 	}
 
 	// Update the VSphereVM's BIOS UUID.
-	ctx.VSphereVM.Spec.BiosUUID = vm.BiosUUID
+	ctx.Logger.Info("vm bios-uuid", "biosuuid", vm.BiosUUID)
```

Comment:
Hi @yastij,

I think this check should validate the `BiosUUID` is non-empty. It should *never* be empty, and thus an error should be returned here, causing the reconcile event to requeue.

Reply:
@akutz - do you mean something like this ?:

```go
	if vm.BiosUUID != "" {
		ctx.VSphereVM.Spec.BiosUUID = vm.BiosUUID
	} else {
         // return err and requeue
         ....
        }